doc/developer-guides/coding_guidelines.rst

```diff
@@ -1156,54 +1473,205 @@ Compliant example::
        char showcase = showcase_array[10];
 
 
+ST-20: The else statement shall not be empty if it is following a else if
+=========================================================================
 
-Expressions
-***********
-
-EP-01: The initialization expression of a for loop shall be simple
-==================================================================
-
-The initialization expression of a for loop shall only be used to initialize the
-loop counter. All other operations shall not be allowed.
+Either a non-null statement or a comment shall be included in the else
+statement. This is to guarantee that the developers have considered all of the
+possible cases.
 
 Compliant example::
 
-    uint32_t i;
+    uint32_t param, showcase;
     
-    for (i = 0U; i < 5U; i++) {
-        printf("count: %d \n", i);
+    if (param < 10U) {
+        showcase = 10U;
+    } else if (param < 20U) {
+        showcase = 20U;
+    } else {
+        showcase = 30U;
     }
 
 .. rst-class:: non-compliant-code
 
    Non-compliant example::
 
-       uint32_t i;
-       uint32_t showcase = 0U;
+       uint32_t param, showcase;
        
-       for (i = 0U, showcase = 10U; i < 5U; i++) {
-           printf("count: %d \n", i);
+       if (param < 10U) {
+           showcase = 10U;
+       } else if (param < 20U) {
+           showcase = 20U;
+       } else {
        }
 
 
-EP-02: The controlling expression of a for loop shall not be empty
-==================================================================
+ST-21: A switch statement shall have the default statement
+==========================================================
+
+This is to guarantee that the developers have considered all of the possible
+cases.
 
 Compliant example::
 
-    uint32_t i;
+    char showcase;
     
-    for (i = 0U; i < 5U; i++) {
-        printf("count: %d \n", i);
-    }
-
-.. rst-class:: non-compliant-code
-
-   Non-compliant example::
-
-       uint32_t i;
-       
-       for (i = 0U; ; i++) {
+    switch (showcase) {
+    case 'a':
+        /* do something */
+        break;
+    case 'A':
+        /* do something */
+        break;
+    default:
+        /* do something */
+        break;
+    }
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       char showcase;
+       
+       switch (showcase) {
+       case 'a':
+           /* do something */
+           break;
+       case 'A':
+           /* do something */
+           break;
+       }
+
+
+ST-22: Every switch clause shall be terminated with a break statement
+=====================================================================
+
```

Comment:
Maybe add an explanation: ``Falling through a case is not allowed.``

This would prohibit writing something like this then:
```
switch (someoption) {
  case 'a':
  case 'b':
  case 'c':
    // do something
    break;
  case 'd':
  case 'e':
    // do something else
    break;
}
```